src/coremods/core_channel/core_channel.cpp

```diff
@@ -148,16 +148,8 @@ class CoreModChannel : public Module, public CheckExemption::EventListener
 	void ReadConfig(ConfigStatus& status) CXX11_OVERRIDE
 	{
 		ConfigTag* optionstag = ServerInstance->Config->ConfValue("options");
-		Implementation events[] = { I_OnCheckKey, I_OnCheckLimit, I_OnCheckChannelBan };
-		if (optionstag->getBool("invitebypassmodes", true))
-			ServerInstance->Modules.Attach(events, this, sizeof(events)/sizeof(Implementation));
-		else
-		{
-			for (unsigned int i = 0; i < sizeof(events)/sizeof(Implementation); i++)
-				ServerInstance->Modules.Detach(events[i], this);
-		}
-
-		joinhook.modefromuser = optionstag->getBool("cyclehostsfromuser");
+		bool bypassmodes = optionstag->getBool("invitebypassmodes", true);
```

Comment:
`getBool()` never throws so you can just set these after the settings that throw without using a temporary.

Reply:
I wasn't sure if the logging on invalid values was going to be replaced with an exception or not

Reply:
And from a readability standpoint it made more sense to me to group all config reads separate from usage of the values.

Reply:
Config errors which are recoverable from (e.g. `getBool()` where it uses the default if the value is not a recognised boolean value) are not really an issue so it doesn't make sense for them to ever throw.

I feel that from a usability standpoint having a comment like `// The config was valid so we can apply it` would be better if necessary.

Reply:
My thinking with `getBool()` was if a user set an invalid value for a boolean flag, allowing it to be applied may lead to unexpected or undesired behavior that wouldn't be obvious at first. It is logged, but that assumes the user reads stdout or the log file (if configured). For cases like that, most applications would generally throw an error. If we don't plan to I can adjust this logic, I just wanted to future-proof in case we did.

Reply:
It is not currently planned for getBool to ever throw in the v3 release.